qiime2/sdk/plugin_manager.py

```diff
@@ -101,43 +122,112 @@ def _integrate_plugin(self, plugin):
             self.formats[name] = record
         self.type_formats.extend(plugin.type_formats)
 
+    def get_semantic_types(self):
+        types = set()
+
+        for plugin in self.plugins.values():
+            for type_record in plugin.types:
+                types.add(type_record)
+
+        return types
+
     # TODO: Should plugin loading be transactional? i.e. if there's
     # something wrong, the entire plugin fails to load any piece, like a
     # databases rollback/commit
 
-    @property
-    def importable_formats(self):
```

Comment:
This will break q2cli, we can either block this PR until it's updated, or we can shim this property to use get_formats

Reply:
Now that this has been largely signed off on, I think it's time to implement the cli interface for this new functionality like we talked about. If you want, I can address the rest of this review then open a cli PR allowing for proper interfacing with the new `get_formats` which would allow for viewing all importable formats among other things obviously. We can also temporarily reinstate the old interface so this can be merged immediately, then that functionality can be removed once the new interface is ready.

Reply:
Since we are up against a release, I would rather just shim these properties, then we can address q2cli at a later time (whenever we want to).